app/models/simulator.rb

```diff
@@ -360,4 +360,29 @@ def simulator_versions
 
     sim_versions.map {|key,val| val['version'] = key; val }.sort_by {|a| a['latest_started_at']}
   end
+
+  require 'csv'
+  def runs_csv(runs = self.runs)
+    run_attr = %w(_id status hostname real_time started_at finished_at)
+    ps_attr = [[:ps,:_id]] + parameter_definitions.map {|pd| [:ps,:v,pd.key] }
+    ps_attr_header = ["ps_id"] + parameter_definitions.map {|pd| "p.#{pd.key}" }
+    latest_run = Run.where(simulator: self, status: :finished).order_by(updated_at: :desc).first
+    result_attr_header = plottable_keys(latest_run&.result).map {|key| "r.#{key}" }
+    result_attr = plottable_keys(latest_run&.result).map {|key| [:result]+key.split('.') }
+    attr = run_attr + ps_attr + result_attr
+    header = run_attr + ps_attr_header + result_attr_header
+
+    aggregated = Run.collection.aggregate([
+      { '$match'  => runs.selector },
```

Comment:
`runs` includes runs with several status like :finished, :submitted, :running and :failed.
Is it better to select runs with status :finished?

Reply:
In my view, it is okay to collect all runs because it is easy for a user to filter out non-finished runs from a CSV file.
In future, we may implement an option to export only "finished" runs.